aggregators/mtg_cards/cards.js

```diff
@@ -1,87 +1,79 @@
-                                cardMap.set(card.name,card);
-                            }
-                        }else{
-                            cardMap.set(card.name,card);
-                        }
-                    });
-                    const cards =  Array.from(cardMap).map(function(element){
-                        return element[1];
-                    });
-                    let card = cards.find(function(card){
-                        return card.name.toLowerCase() === cardName;
-                    });
-                    if(!card){
-                        card = cards.find(function (card) {
-                           return card.name.toLowerCase().startsWith(cardName);
-                        })
-                    }
-                    if(!card){
-                        card = cards[0];
-                    }
-                    const cardInfo = ["**" + card.name + "**"];
-                    if (card.type) {
-                        cardInfo.push(card.type);
-                    }
-                    if (card.text) {
-                        cardInfo.push(card.text.replace(/\*/g, '\\*'));
-                    }
-                    if (card.loyalty) {
-                        cardInfo.push(card.loyalty);
-                    }
-                    if (card.power) {
-                        cardInfo.push(card.power.replace(/\*/g, '\\*') + "/" + card.toughness.replace(/\*/g, '\\*'));
-                    }
-                    if (card.printings) {
-                        cardInfo.push(card.printings.join(", "));
-                    }
-                    cardInfo.push("http://magiccards.info/query?q=!" + encodeURIComponent(card.name));
-                    if (card.imageUrl) {
-                        cardInfo.push(card.imageUrl);
-                    }
-                    callback(cardInfo.join("\n"));
-                    if(cards.length>1){
-                        const cardNames = [];
-                        cards.forEach(function(element){
-                            if(!(element.name===card.name)){
-                                cardNames.push((element.name));
-                            }
-                        });
-                        callback("Other matching cards are: " + cardNames.join(", "))
-                    }
+            url: this.cardApi + cardName,
+            json: true
+        },
+        (error, response, body) => {
+            if (!error && response.statusCode === 200 && body.cards && body.cards.length) {
+                const card = this.findCard(cardName, body.cards);
+                let response = this.cardToString(card);
+                let otherCardNames = body.cards.map(c => c.name).filter(n => n !== card.name);
+                if (otherCardNames.length) {
+                    otherCardNames.sort();
+                    otherCardNames = _.sortedUniq(otherCardNames);
+                    response += "\n\nOther matching cards: " + otherCardNames.join(", ");
```

Comment:
It would be better to `join` the card names with `join(" | ")` because some card names contain commas.